src/types.ts

```diff
@@ -75,6 +75,7 @@ export interface UnpluginOptions {
     onResolveFilter?: RegExp
     onLoadFilter?: RegExp
     setup?: EsbuildPlugin['setup']
+    loader?: Loader
```

Comment:
Maybe `loader` should be a function that is processed by userland, instead of a constant value.

Reply:
```suggestion
    loader?: () => Loader
```

Reply:
```suggestion
    loader?: () => Loader
```

Reply:
That makes a lot of sense. Would you like to make it `Loader | ((code:string, id: string) => Loader)` for convenience or should I stick with your suggestion?

Reply:
It would be also great